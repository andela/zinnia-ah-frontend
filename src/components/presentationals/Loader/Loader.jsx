import React from 'react';
import PropTypes from 'prop-types';

const Loader = ({ text, size }) => {
  return (
    <div className="ui active inverted dimmer">
      <div className={`ui indeterminate text loader ${size}`}>{text}</div>
    </div>
  );
};

Loader.propTypes = {
  size: PropTypes.string,
  text: PropTypes.string,
};

export default Loader;
