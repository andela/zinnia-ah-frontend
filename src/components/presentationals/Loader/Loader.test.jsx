import React from 'react';
import { shallow } from 'enzyme';

import Loader from './Loader';

const text = 'signing up';
const size = 'large';

const component = shallow(<Loader text={text} size={size} />);

describe('<Loader />', () => {
  it('mounts without exploding', () => {
    expect(component).toMatchSnapshot();
  });

  it('should receive the correct props', () => {
    const { className, children } = component.props().children.props;

    expect(className).toContain(size);
    expect(children).toBe(text);
  });
});
