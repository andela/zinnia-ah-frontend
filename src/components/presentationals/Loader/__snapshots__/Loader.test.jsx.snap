// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> mounts without exploding 1`] = `
<div
  className="ui active inverted dimmer"
>
  <div
    className="ui indeterminate text loader large"
  >
    signing up
  </div>
</div>
`;
